src/System.Text.Primitives/System/Text/Encoders/SequenceValidity.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Represents the validity of a UTF code unit sequence.
+    /// </summary>
+    public enum SequenceValidity
+    {
+        /// <summary>
+        /// The sequence is empty.
+        /// </summary>
+        Empty = 0,
```

Comment:
Why do we need both Empty and WellFormed?

Reply:
This enum is only used for the "read the first sequence from the string and tell me about it" API. The caller needs to distinguish between empty string and valid string so that he doesn't try interpreting the [meaningless] first sequence in an empty string. I thought about making these APIs "TryPeek" instead, but having both a bool and an enum returned by a single method left a bad taste in my mouth.

BTW, the "is this string valid?" APIs simply return bool, not this enum. This enum is only used when the caller needs to perform deep inspection, such as if he's iterating or performing replacement.